src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,184 @@
+                String.format("User should be added with age %d", VALID_AGE + 1));
+    }
+
+    @Test
+    void registerEmptyPassword_NotOk() {
+        invalidUser.setPassword("");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(invalidUser),
+                String.format("User should not be added with empty password %s", EXCEPTION));
+    }
+
+    @Test
+    void registerNullPassword_NotOk() {
+        invalidUser.setPassword(null);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(invalidUser),
+                String.format("User should not be added with null password %s", EXCEPTION));
+    }
+
+    @Test
+    void registerTooShortPassword_NotOk() {
+        String newPassword = "1234";
+        invalidUser.setPassword(newPassword);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(invalidUser),
+                String.format("User should not be added with %s password %s",
+                        newPassword, EXCEPTION));
+    }
+
+    @Test
+    void registerLongPassword_Ok() {
+        String newPassword = "123456789";
+        validUser.setPassword(newPassword);
+        registrationService.register(validUser);
+        User actual = storageDao.get(validUser.getLogin());
+        Assertions.assertEquals(validUser, actual,
+                String.format("User should be added with %s password", newPassword));
+    }
+
+    @Test
+    void registerManyValidUsers_Ok() {
+        User alise = new User();
+        alise.setLogin(VALID_SECOND_LOGIN);
+        alise.setPassword(VALID_PASSWORD);
+        alise.setAge(VALID_AGE);
+        User john = new User();
+        john.setLogin(VALID_THIRD_LOGIN);
+        john.setPassword(VALID_PASSWORD);
+        john.setAge(VALID_AGE);
+        registrationService.register(alise);
+        registrationService.register(john);
+        registrationService.register(validUser);
+        User actual = storageDao.get(validUser.getLogin());
+        Assertions.assertEquals(validUser, actual,
+                "Method should be added all users to storage, and return last");
+    }
+
+    @Test
+    void registerAndGetInvalidUser_NotOk() {
+        User alise = new User();
+        alise.setLogin(VALID_SECOND_LOGIN);
+        alise.setPassword(VALID_PASSWORD);
+        alise.setAge(VALID_AGE);
+        User john = new User();
+        john.setLogin(VALID_THIRD_LOGIN);
+        john.setPassword(VALID_PASSWORD);
+        john.setAge(VALID_AGE);
+        registrationService.register(alise);
+        registrationService.register(john);
+        User actual = storageDao.get(validUser.getLogin());
+        Assertions.assertNull(actual,
+                "Method should be returned null for invalid get login");
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+}

```

Comment:
you should test all edge case scenarios: password length 5/6/7 and age 17/18/19

also tests naming should be greaterThenValid / shorterThenExpected